### ISSUE TITLE
[BUG/ISSUE] "-lblas -llapack" breaks compilation with ifort/gfortran (GCHP 12.5.0)

### ISSUE THREAD

**Issue by JiaweiZhuang** [geoschem/gchp_legacy#39]
**Describe the bug**

New MAPL uses `-lblas -llapack` in [Shared/Config/math.mk](https://github.com/geoschem/gchp/blob/90f24d9aeab58c988da1f88263ce7b471f799367/Shared/Config/math.mk) (0f050c12c048a3449f4d7fe9c47e58c3921b398a). This breaks the build when using Intel compiler: 

```
/home/centos/gchp_12.5.0/Code.GCHP/GCHP/Shared/Linux/lib/libGMAO_gfio.a /home/centos/gchp_12.5.0/Code.GCHP/GCHP/Shared/Linux/lib/libGMAO_eu.a -L/home/centos/spack/opt/spack/linux-centos7-x86_64/intel-19.0.4/netcdf-fortran-4.4.5-feoqwqgpgo6l3wyozlqj4leesuhnddjw/lib -lnetcdff -L/home/centos/spack/opt/spack/linux-centos7-x86_64/intel-19.0.4/netcdf-4.7.0-5xnzracbnatiq32kh5vukohqckxqtacj/lib -lnetcdf -lnetcdf -lhdf5_hl -lhdf5 -lz -lm -L/home/centos/spack/opt/spack/linux-centos7-x86_64/intel-19.0.4/netcdf-4.7.0-5xnzracbnatiq32kh5vukohqckxqtacj/lib -lnetcdf -lblas -llapack -ldl -lc -lpthread -lrt  -L/usr/lib/gcc/x86_64-redhat-linux/4.8.5 -lstdc++
ld: cannot find -llapack
ld: cannot find -llapack
ld: cannot find -llapack
ld: cannot find -llapack
```

Full logs (the error is independent of MPI):
- [compile_ifort_openmpi_fail_at_blas.log](https://github.com/geoschem/gchp/files/3596925/compile_ifort_openmpi_fail_at_blas.log)
- [compile_ifort_intelmpi_fail_at_blas.log](https://github.com/geoschem/gchp/files/3596926/compile_ifort_intelmpi_fail_at_blas.log)

With ifort, the Makefile should in principle pick up MKL instead of BLAS/LAPACK:
https://github.com/geoschem/gchp/blob/90f24d9aeab58c988da1f88263ce7b471f799367/Shared/Config/math.mk#L18-L21

But in fact it is still linking LAPACK: 
https://github.com/geoschem/gchp/blob/90f24d9aeab58c988da1f88263ce7b471f799367/Shared/Config/math.mk#L39-L45

**To Reproduce**
1. Install Intel compiler 19.0.4 following [Spack documentation](https://spack.readthedocs.io/en/latest/build_systems/intelpackage.html)
2. Install NetCDF via Spack by `spack spec netcdf-fortran %intel ^netcdf~mpi ^hdf5~mpi+fortran+hl` (independent of MPI installation). OpenMPI can be installed via `spack install openmpi+pmi schedulers=slurm % intel`. Intel MPI is pre-installed on the AWS cluster.
3. Build GCHP 12.5.0. The environment config is the same as #37, plus `export gFTL=...`.

**Environment**
- GEOS-Chem/GCHP version: 12.5.0
- Compiler version: ifort 19.0.4
- MPI library and version: OpenMPI 3.1.4 and Intel MPI 19.0.4
- netCDF and netCDF-Fortran library version: 4.7.0 and 4.4.5
- Computational environment: AWS cloud, CentOS 7
- Code modifications: No change to GC-classic part; apply #35 to GCHP part

**Comment by JiaweiZhuang** [comment]
~GNU compiler does not have this problem~ (but has another problem #40)

**Comment by JiaweiZhuang** [comment]
It turns out that gfortran also has the exact same problem. The quickest way to fix it is `yum install lapack-devel blas-devel`. Although this leads to warning that
```
/usr/bin/ld: warning: libgfortran.so.3, needed by /lib/../lib64/libblas.so, may conflict with libgfortran.so.5
/usr/bin/ld: warning: libgfortran.so.3, needed by /lib/../lib64/libblas.so, may conflict with libgfortran.so.5
```
which is presumably due to different gfortran versions (gcc 8.3.0 from Spack and gcc 4.8.5 on the default system).

**Comment by JiaweiZhuang** [comment]
@lizziel Do you have a successful compile log for GCHP 12.5.0 with ifort on Odyssey? I wonder whether it should link MKL or BLAS... I saw no `-mkl` flags when compiling with ifort.

**Comment by lizziel** [comment]
This should be fixed in the upcoming 13.0.0 release which retires the GNU makefiles. If there is a still a problem when using CMake it should be created as a new issue.
